src/cactus/update/cactus_update_prepare.py

```diff
@@ -0,0 +1,382 @@
+
+            if not os.path.isfile(path):
+                raise RuntimeError(
+                    "Invalid path for the sequence name \'{}\': {}".format(name, path))
+
+            inFasta[name] = (weight, os.path.abspath(path))
+
+    return inFasta
+
+
+def main():
+    #print("children: {}".format(halStats_call_get_children("/Users/thiagogenez/Documents/buckets/adding-genome/lepidoptera/runs/0/steps/Anc05.hal", "Anc05")))
+
+    # Same main parser as usual
+    parser = ArgumentParser(formatter_class=ArgumentDefaultsHelpFormatter,
+                            usage='%(prog)s {node,branch} [-h] [Options]', add_help=False)
+    parser._positionals.title = "subcommand"
+
+    # Same subparsers as usual
+    subparser = parser.add_subparsers(
+        help='Desired alignment update approach', dest='action')
+
+    # Usual arguments which are applicable for the whole script / top-level args
+    # parser.add_argument('--verbose', help='Common top-level parameter',
+    #                    action='store_true', required=False)
+
+    # hack to add --help
+    parser.add_argument('--help', '-h', action="store_true",
+                        dest='help', help=SUPPRESS)
+
+    # Create parent subparser. Note `add_help=False` and creation via `argparse.`
+    parent_parser = ArgumentParser(
+        formatter_class=ArgumentDefaultsHelpFormatter, add_help=False)
+    parent_parser.add_argument(
+        "inHal", help="The input HAL-format file containing the existing alignment")
+    parent_parser.add_argument(
+        "inFasta", help="Tab-separated file. First column: genome name, second column: FASTA-format file path")
+
+    # taken from cactus-prepare
+    parent_parser.add_argument(
+        "--halOptions", type=str, default="--hdf5InMemory", help="options for every hal command")
+    parent_parser.add_argument(
+        "--outDir",  default='.', help='Directory where assemblies and cactus-call dependencies will be placed.', dest='work_dir')
+    parent_parser.add_argument('--skip-backup', action='store_true', help='Skip the backup of the given HAL file', dest='skip_backup')
+    parent_parser.add_argument(
+        "--skip-halValidate", help="Skip the validation of the given HAL file", action="store_false", dest='skip_halValidate')
+    parent_parser.add_argument("--cactus-prepare-options", dest='cactus_prepare_options',
+                               type=str, default="--preprocessBatchSize 1 --cactusOptions '--realTimeLogging --logInfo --retryCount 0'", help="Options to bypass local configuration for cactus-prepare")
+
+    # add subcommands options
+    add_subcommand_options(subparser, parent_parser, 'node')
+    add_subcommand_options(subparser, parent_parser, 'branch')
+
+    options = parser.parse_args()
+
+    if not options.action:
+        parser.print_help()
+        sys.exit(1)
+
+    # validate the given inFasta text file
+    options.inFasta = inFasta_sanity_check(options.inFasta)
```

Comment:
No, the `inFile` is not the usual `SeqFile`. It is just a  list of genomes to be included.  I will rename this variable to avoid further confusion.

I need to improve how the user will provide this piece of information to update its alignment. Any suggestion about this @glennhickey ? From a user point of view, how you would like to provide this piece of information to get your alignment updated?